### PR TITLE
ETCD-478: Use golang etcd client for auto backups

### PR DESCRIPTION
**Pull request by tjungblu**

### CODE REVIEW

**Comment by tjungblu** [comment]
seems to have some issues connecting to etcd locally:

```
[tjungblu ~]$ kubectl logs -n openshift-etcd cluster-backup-job-ptvll-dj2t6
Defaulted container "cluster-backup" out of: cluster-backup, verify-storage (init)
E0814 11:07:39.201629       1 backuprestore.go:66] run: backup failed: backup: failed to get etcd client: failed to make etcd client for endpoints [127.0.0.1:2379]: context deadline exceeded
I0814 11:07:39.201664       1 backuprestore.go:68] config-dir is: /etc/kubernetes
```

File: bindata/etcd/cluster-backup-job.yaml [hunk 1]
```diff
@@ -49,7 +49,12 @@ spec:
               name: static-pod-dir
             - mountPath: /etc/kubernetes/cluster-backup
               name: etc-kubernetes-cluster-backup
+            - mountPath: /var/run/secrets/etcd-client
```
Comment (Elbehery) [inline bindata/etcd/cluster-backup-job.yaml:60]: i wonder how was it working before then ?

Comment (tjungblu) [reply bindata/etcd/cluster-backup-job.yaml:60]: it's coded into the script itself
https://github.com/openshift/cluster-etcd-operator/blob/master/bindata/etcd/cluster-backup.sh#L110-L112

Comment (Elbehery) [reply bindata/etcd/cluster-backup-job.yaml:60]: achsooo :D :D

**Comment by Elbehery** [comment]
looks like the failure is coming from monitoring stack 

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-etcd-operator/1099/pull-ci-openshift-cluster-etcd-operator-master-e2e-aws-ovn-etcd-scaling/1691379050347499520

**Comment by tjungblu** [comment]
slated for 4.15 after branching on Thursday. Todo: squash and give it a better commit message.

**Pull request merged**